src/app/shared/select-edit-value.component.ts

```diff
@@ -0,0 +1,109 @@
+const selectEditValueLayout: HTMLTemplateElement = document.createElement('template');
+selectEditValueLayout.innerHTML = `
+	<div class="d-flex align-items-center justify-content-between gap-2">
+		<button type="button" class="btn btn-icon btn-primary">
+			<span class="visually-hidden"></span>
+			<app-icon data-name="Minus_small"></app-icon>
+		</button>
+		<output></output>
+		<button type="button" class="btn btn-icon btn-primary">
+			<span class="visually-hidden"></span>
+			<app-icon data-name="Plus_small"></app-icon>
+		</button>
+	</div>
+`;
+
+class SelectEditValueComponent extends HTMLElement {
+	static observedAttributes = ['data-name', 'data-index', 'data-setting-values'];
+	selectedValue: HTMLOutputElement | null = null;
+	btnPrevValue: HTMLButtonElement | null = null;
+	btnNextValue: HTMLButtonElement | null = null;
+
+	name = '';
+	values: string[] = [];
+	currentIndex: number = null;
+	currentValue: string = null;
+
+	handler: any;
+
+	constructor() {
+		super();
+
+		this.name = this.dataset?.name || this.name;
+
+		this.appendChild(selectEditValueLayout.content.cloneNode(true));
+
+		this.handler = this.createHandler();
+	}
+
+	connectedCallback(): void {
+		this.selectedValue = this.querySelector('output');
+		this.btnPrevValue = this.querySelector('* > button:first-child');
+		this.btnNextValue = this.querySelector('* > button:last-child');
```

Comment:
Les `* >`  au début du sélecteur ne sont pas utiles je pense.
De plus, j’utilsierais plutôt `:first-of-type` et `:last-of-type`, ça rendrait le sélecteur un peu plus robuste à d’éventuels changements dans le DOM.